include/zephyr/drivers/kscan_matrix.h

```diff
@@ -0,0 +1,166 @@
+/*
```

Comment:
why isn't this _new_ API part of `kscan.h`?

Reply:
It is requested during the [review conversation](https://github.com/zephyrproject-rtos/zephyr/pull/42923#discussion_r849667743).